builder/chroots.go

```diff
@@ -0,0 +1,637 @@
+	VersionURL   string
+	// Format is already in b.Format.
+}
+
+// TODO: Move this to the more general configuration handling.
+func readBuildChrootsConfig(path string) (*buildChrootsConfig, error) {
+	iniFile, err := ini.InsensitiveLoad(path)
+	if err != nil {
+		return nil, err
+	}
+
+	cfg := &buildChrootsConfig{}
+
+	// TODO: Validate early the fields we read.
+	server, err := iniFile.GetSection("Server")
+	if err == nil {
+		cfg.HasServerSection = true
+		cfg.DebugInfoBanned = server.Key("debuginfo_banned").Value()
+		cfg.DebugInfoLib = server.Key("debuginfo_lib").Value()
+		cfg.DebugInfoSrc = server.Key("debuginfo_src").Value()
+	}
+
+	swupd, err := iniFile.GetSection("swupd")
+	if err != nil {
+		return nil, fmt.Errorf("error in configuration file %s: %s", path, err)
+	}
+
+	getKey := func(section *ini.Section, name string) (string, error) {
+		key, kerr := section.GetKey(name)
+		if kerr != nil {
+			return "", fmt.Errorf("error in configuration file %s: %s", path, kerr)
+		}
+		return key.Value(), nil
+	}
+
+	cfg.UpdateBundle, err = getKey(swupd, "BUNDLE")
+	if err != nil {
+		return nil, err
+	}
+	cfg.ContentURL, err = getKey(swupd, "CONTENTURL")
+	if err != nil {
+		return nil, err
+	}
+	cfg.VersionURL, err = getKey(swupd, "VERSIONURL")
+	if err != nil {
+		return nil, err
+	}
+
+	return cfg, nil
+}
+
+func (b *Builder) buildBundleChroots(set *bundleSet) error {
+	var err error
+	stateDir := b.Statedir
+	chrootDir := filepath.Join(stateDir, "image")
+
+	// TODO: Remove remaining references to outputDir. Let "build update" take care of
+	// bootstraping or cleaning up.
+	outputDir := filepath.Join(stateDir, "www")
+
+	if stateDir == "" || stateDir == "/" {
+		return errors.Errorf("invalid state dir: %s", stateDir)
+	}
+
+	if _, ok := set.Bundles["os-core"]; !ok {
+		return fmt.Errorf("os-core bundle not found")
+	}
+
+	// Bootstrap the directories.
+	err = os.MkdirAll(filepath.Join(chrootDir, "0"), 0755)
+	if err != nil {
+		return err
+	}
+	err = os.MkdirAll(filepath.Join(outputDir, "0"), 0755)
+	if err != nil {
+		return err
+	}
+	// TODO: Move this outside this function. chroot building doesn't use LAST_VER.
+	lastVerPath := filepath.Join(chrootDir, "LAST_VER")
+	if _, err = os.Stat(lastVerPath); os.IsNotExist(err) {
+		err = ioutil.WriteFile(lastVerPath, []byte("0\n"), 0644)
+		if err != nil {
+			return err
+		}
+	}
+
+	// TODO: To not touch config code that is in flux at the moment, reparsing it here to grab
+	// information that previously Mixer didn't care about. Move that to the configuration part
+	// of Mixer.
+	cfg, err := readBuildChrootsConfig(b.Buildconf)
+	if err != nil {
+		return err
+	}
+
+	if _, ok := set.Bundles[cfg.UpdateBundle]; !ok {
+		return fmt.Errorf("couldn't find bundle %q specified in configuration as the update bundle", cfg.UpdateBundle)
+	}
+
+	// Write INI files. These are used to communicate to the next step of mixing (build update).
+	var serverINI bytes.Buffer
+	fmt.Fprintf(&serverINI, `[Server]
```

Comment:
This is a good use case for a template.

Reply:
Maybe, but since we are exploring how to use information directly from builder.conf, I have the impression it will go away, so won't take the trouble. If it does stay, we templatize it.

Created https://github.com/clearlinux/mixer-tools/issues/114 to keep track of server.ini's fate.